tiledb/sm/array/array.cc

```diff
@@ -211,12 +211,31 @@ Status Array::open(
   RETURN_NOT_OK(
       encryption_key_.set_key(encryption_type, encryption_key, key_length));
 
-  timestamp_ =
-      (query_type == QueryType::READ) ? utils::time::timestamp_now_ms() : 0;
   metadata_.clear();
   metadata_loaded_ = false;
   non_empty_domain_computed_ = false;
 
+  bool found = false;
+  uint64_t timestamp_cfg = 0;
+  RETURN_NOT_OK(config_.get<uint64_t>(
+      "sm.array.open_timestamp_start", &timestamp_cfg, &found));
+  assert(found);
+
+  if (query_type == QueryType::READ) {
+    if (timestamp_cfg == UINT64_MAX) {
+      timestamp_ = utils::time::timestamp_now_ms();
+    } else {
+      timestamp_ = timestamp_cfg;
```

Comment:
It may be a good idea to rename `timestamp_` to `timestamp_end_` here, as later we will need a `timestamp_start_` for the corresponding start timestamp.